tests/src/JIT/Performance/CodeQuality/Span/Indexer.cs

```diff
@@ -0,0 +1,925 @@
+        public static void WriteViaIndexer1(int length)
+        {
+            byte[] a = GetData(length);
+
+            Invoke((int innerIterationCount) =>
+            {
+                Span<byte> s = new Span<byte>(a);
+                byte result = 0;
+                for (int i = 0; i < innerIterationCount; ++i)
+                {
+                    result = WriteViaIndexer1(s);
+                }
+                return result;
+            },
+            "WriteViaIndexer1({0})", length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte WriteViaIndexer1(Span<byte> data)
+        {
+            byte q = data[0];
+
+            for (var idx = 1; idx < data.Length; idx++)
+            {
+                data[0] ^= data[idx];
+            }
+
+            byte x = data[0];
+            data[0] = q;
+
+            return x;
+        }
+
+        [Benchmark(InnerIterationCount = Iterations)]
+        [InlineData(DefaultLength)]
+        [Category("Indexer in-loop bounds check elimination w/ writes")]
+        public static void WriteViaIndexer2(int length)
+        {
+            byte[] a = GetData(length);
+
+            Invoke((int innerIterationCount) =>
+            {
+                Span<byte> s = new Span<byte>(a);
+                byte result = 0;
+                for (int i = 0; i < innerIterationCount; ++i)
+                {
+                    result = WriteViaIndexer2(s);
+                }
+                return result;
+            },
+            "WriteViaIndexer2({0})", length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte WriteViaIndexer2(Span<byte> data)
+        {
+            byte x = data[0];
+
+            for (var idx = 1; idx < data.Length; idx++)
+            {
+                // Bounds checks are redundant
+                byte b = data[idx];
+                x ^= b;
+                data[idx] = b;
+            }
+
+            return x;
+        }
+
+        [Benchmark(InnerIterationCount = Iterations)]
+        [InlineData(DefaultLength)]
+        [Category("Span known size bounds check elimination")]
+        public static void KnownSizeArray(int length)
+        {
+            if (length != 1024)
+            {
+                throw new Exception("test requires 1024 byte length");
+            }
+
+            Invoke((int innerIterationCount) =>
+            {
+                byte result = TestKnownSizeArray(innerIterationCount);
+                return result; 
+            },
+            "KnownSizeArray({0})", length);
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static byte TestKnownSizeArray(int innerIterationCount)
+        {
+            byte[] a = new byte[1024];
+            SetData(a);
+            Span<byte> data = new Span<byte>(a);
+            byte x = 0;
+
+            for (int i = 0; i < innerIterationCount; i++)
+            {
```

Comment:
I may have misplaced the comment; I was curious in `KnownSizeArray` and friends why not generate a new array per inner iteration.  (not that it's particularly concerning, just curious since I noticed it departing from the pattern of inner iteration counting calls to the kernel method)